### ISSUE TITLE
NullPointerException calling Object.equals(Object) for two Integer objects

### ISSUE THREAD

**Issue by hzongaro** [eclipse/openj9#7584]
The following program often yields a `NullPointerException`:

```
class Foo {
  public static final void main(String args[]) {
    long t1 = System.currentTimeMillis();
    for (int i = 0; i < 100000; i++) {
      run(new Integer(i));
    }
    long t2 = System.currentTimeMillis();
    System.out.println("Time == "+(t2-t1)+"ms");
  }

  public static final int run(Object o) {
    int sum = 0;
    for (int i = 0; i < 100000; i++) {
      Object y = new Integer(i);
      if (y.equals(o)) sum++;
    }
    return sum;
  }
}
```

when run with the option `-Xjit:{Foo.main*}(dontInline={Foo.run*})`

**Comment by hzongaro** [comment]
The problem lies in Escape Analysis.  With `traceEscapeAnalysis` for `Foo.run`, we see this at veryHot optimization level:

```
Final candidates
Candidate 0:
   Node = 00007FC77D01FA30, contiguous = 0, local = 1
   Value numbers = { 12 }
   2 fields:
      0: offset=8   size=4  vectorElem=0  symRef=null good={#384} bad={}
      1: offset=0   size=4  vectorElem=0  symRef=null good={#295} bad={}
Change node [00007FC77D0C2850] into a direct load or store of #414 (4 bytes) field 0 cand 00007FC77E10BB10
Change node [00007FC77D0C2BC0] into a direct load or store of #415 (8 bytes) field 1 cand 00007FC77E10BB10
Change node [00007FC77D0C31B0] into a direct load or store of #414 (4 bytes) field 0 cand 00007FC77E10BB10
Change node [00007FC77D0BC630] into a direct load or store of #414 (4 bytes) field 0 cand 00007FC77E10BB10
Change node [00007FC77D0BF1A0] into a direct load or store of #415 (8 bytes) field 1 cand 00007FC77E10BB10
Change node [00007FC77D0BDE40] into a direct load or store of #414 (4 bytes) field 0 cand 00007FC77E10BB10
Change node [00007FC77D0C4A10] into a direct load or store of #414 (4 bytes) field 0 cand 00007FC77E10BB10
Pass: (0) Non-contiguous allocation found in Foo.run(Ljava/lang/Object;)I
```

Before that pass of EA, we have

```
n450n     ifacmpne --> block_29 BBStart at n260n (inlineProfiledGuard )                       [0x7fc77d0c2c60] bci=[2,0,986] rc=0 vc=315 vn=79 li=- udi=- nc=2 flg=0x1020
n451n       aloadi  <vtable-entry-symbol>[#398  Shadow +376] [flags 0x10607 0x0 ]             [0x7fc77d0c2cb0] bci=[-1,19,15] rc=1 vc=315 vn=17 li=- udi=- nc=1
n448n         aloadi  <vft-symbol>[#295  Shadow] [flags 0x18607 0x0 ] (X!=0 X>=0 sharedMemory )  [0x7fc77d0c2bc0] bci=[-1,19,15] rc=1 vc=315 vn=16 li=- udi=- nc=1 flg=0x104. <<<
n449n           aload  <auto slot 3>[#364  Auto] [flags 0x7 0x0 ] (X!=0 X>=0 sharedMemory )   [0x7fc77d0c2c10] bci=[-1,19,15] rc=1 vc=315 vn=12 li=49 udi=41 nc=0 flg=0x104
n452n       aconst 0xab3fb0 (methodPointerConstant X!=0 sharedMemory )                        [0x7fc77d0c2d00] bci=[2,0,986] rc=1 vc=315 vn=15 li=- udi=- nc=0 flg=0x2004
```

and after,

```
n450n     ifacmpne --> block_29 BBStart at n260n (inlineProfiledGuard )                       [0x7fc77d0c2c60] bci=[2,0,986] rc=0 vc=315 vn=- li=- udi=- nc=2 flg=0x1020
n451n       aloadi  <vtable-entry-symbol>[#398  Shadow +376] [flags 0x10607 0x0 ]             [0x7fc77d0c2cb0] bci=[-1,19,15] rc=1 vc=315 vn=- li=- udi=- nc=1
n448n         aload  <temp slot 8>[#415  Auto] [flags 0x20000007 0x0 ] (X!=0 X>=0 sharedMemory )  [0x7fc77d0c2bc0] bci=[-1,19,15] rc=1 vc=315 vn=- li=- udi=- nc=0 flg=0x104
n452n       aconst 0xab3fb0 (methodPointerConstant X!=0 sharedMemory )                        [0x7fc77d0c2d00] bci=[2,0,986] rc=1 vc=315 vn=- li=- udi=- nc=0 flg=0x2004
```

Notice that the reference to `vft-symbol` is replaced with the temporary field, `#415`.  But that temporary is only ever initialized with a null pointer, which ultimately leads to the `NullPointerException`.

```
n629n     astore  <temp slot 8>[#415  Auto] [flags 0x20000007 0x0 ]                           [0x7fc77d0c6450] bci=[-1,10,14] rc=0 vc=0 vn=- li=- udi=- nc=1
n628n       aconst NULL                                                                       [0x7fc77d0c6400] bci=[-1,10,14] rc=1 vc=0 vn=- li=- udi=- nc=0
```

**Comment by hzongaro** [comment]
This bug was introduced by pull request #4282, which was a fix for issue #4281.
